ci-operator/step-registry/ipi/conf/azure/ipi-conf-azure-commands.sh

```diff
@@ -22,12 +22,20 @@ elif [[ "${SIZE_VARIANT}" == "compact" ]]; then
   master_type=Standard_D8s_v3
 fi
 
+platform_azure_opt={
+  baseDomainResourceGroupName: os4-common
+  region: ${REGION}}
+
+if [ X"${OUTBOUND_TYPE}" == X"UserDefinedRouting" ]; then
+  echo "Adding 'outboundType: UserDefinedRouting' in install-config.yaml"
+  platform_azure_opt=platform_azure_opt.merge({"outboundType": ${outboundType}})
+fi
+
 cat >> "${CONFIG}" << EOF
 baseDomain: ${BASE_DOMAIN}
 platform:
   azure:
-    baseDomainResourceGroupName: os4-common
-    region: ${REGION}
+    platform_azure_opt
```

Comment:
instal-config.yaml never has such a key - `platform.azure.platform_azure_opt`, `platform_azure_opt` is only a var name only existing in flexy-template ruby script. It is a map collections defined in ruby, only recognized by flexy-template ruby script, you have to find out a way to assemble install-config.yaml in the shell syntax. Forget the syntax and definition in flexy-template ruby script. It does not work to copy those ruby code here.

Reply:
Have changed to implement if statement. Let me know if there is a better way to implement it